### ISSUE TITLE
Using `ItemGroup`, `PropertyGroup` (etc) suggestions from the extension results in incorrect indentation

### ISSUE THREAD

**Issue by alexrp** [tintoy/msbuild-project-tools-vscode#93]
Let's say I have a file like this:

```xml
<Project>
</Project>
```

If I start to type an item group like

```xml
<Project>
    <Item
</Project>
```

and accept the `<ItemGroup>` suggestion from the extension, I end up with:

```xml
<Project>
 <ItemGroup>
   
 </ItemGroup>
```

This is with `indent_size = 2` in `.editorconfig`. With `indent_size = 4`, it instead ends up as:

```xml
<Project>
   <ItemGroup>
       
   </ItemGroup>
```

**Comment by tintoy** [comment]
Hi, sorry for the late reply - I didn't see the notification email for this until this morning :)

Just checking if I understand correctly, are you saying that it appears to be using the indent size specified in settings (in this case, from `.editorconfig`), rather than the indent level (i.e. column) where you started typing?

**Comment by alexrp** [comment]
It seems to respect both the current indent level and `indent_size` from `.editorconfig`, *but* it's always off by one space.

**Comment by tintoy** [comment]
Sorry, I can see that'd be annoying. I'll look into it and let you know what I find 🙂

**Comment by RReverser** [comment]
This is still an issue. Did you ever get time to investigate it?

**Comment by tintoy** [comment]
Hi - sorry, I remember investigating, but I can’t remember what I found. I’ll have another look and see if there is anything obvious.

just as a quick check, does this happen when the file’s language is set to MSBuild as well as when it is set to XML?

**Comment by RReverser** [comment]
I don't think this extension's autocomplete even works when set to XML? At least I only tried 2ith language set to MSBuild.

**Comment by tintoy** [comment]
Ok, thanks - that helps to cut down on the possibilities :)

**Comment by tintoy** [comment]
I wonder if this could be related to tintoy/msbuild-project-tools-server#71…

**Comment by tintoy** [comment]
I’m still looking into this, but no luck so far. I suspect it maybe be related to how we compute the span of text to be replaced by the completion but it’s tricky to test for.

**Comment by RReverser** [comment]
This kind of off-by-one error almost seems like something is mixing up 1-based column numbers with 0-based ones.

**Comment by tintoy** [comment]
While it isn’t impossible, everything uses the same data structures to represent LSP line-and-column vs internal line-and-column (and the translations are always done by the same function), it so may not be all that likely (if that was broken it would be broken everywhere). Worth checking though, so I’ll do that, thanks 🙂

**Comment by tintoy** [comment]
Sorry for the delay, I’ll put together a new release shortly.

**Comment by tintoy** [comment]
Published [v0.6.5](https://github.com/tintoy/msbuild-project-tools-vscode/releases/tag/v0.6.6) (it may take up to 15-20 minutes before the new version is visible in the VS extension gallery).

@RReverser - if you have a moment, would you be able to give the new version a try and see if it fixes the problem for you?

**Comment by RReverser** [comment]
Looks like there's a different issue now: I'm typing `<`, then selecting any of the autocomplete tags, e.g. `Analyzer`. When I press Enter, I'm getting `<<Analyzer Include="" />` - that is, the original `<` is not replaced.

**Comment by tintoy** [comment]
Is your language set to XML or MSBuild?

**Comment by RReverser** [comment]
Oops, yeah, I changed them to XML while this extension's autocomplete was broken. Seems to work with MSBuild.

Why does the language affect such subtle detail?

**Comment by tintoy** [comment]
This bug should be fixed as of [v0.6.6](https://github.com/tintoy/msbuild-project-tools-vscode/releases/tag/v0.6.6), at least when language is set to `MSBuild`.
